cmd/atlas/internal/client/driver.go

```diff
@@ -0,0 +1,109 @@
+package client
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"ariga.io/atlas/sql/mysql"
+	"ariga.io/atlas/sql/postgres"
+	"ariga.io/atlas/sql/schema"
+	"ariga.io/atlas/sql/sqlite"
+)
+
+type (
+	// Driver provides to diff, execute and inspect against a Data Source.
+	Driver interface {
+		schema.Differ
+		schema.Execer
+		schema.Inspector
+	}
+	// AtlasDriver implements the Driver interface using Atlas.
+	AtlasDriver struct {
+		schema.Differ
+		schema.Execer
+		schema.Inspector
+	}
+
+	dbName string
+)
+
+const (
+	mysqlDB    dbName = "mysql"
+	postgresDB dbName = "postgres"
+	sqliteDB   dbName = "sqlite3"
+)
+
+var providers = map[dbName]func(string) (*AtlasDriver, func(), error){
+	mysqlDB:    atlasDriverMysql,
+	postgresDB: atlasDriverPostgres,
+	sqliteDB:   atlasDriverSqlite,
+}
+
+// NewAtlasDriver connects a new Atlas Driver returns AtlasDriver and a closer.
+func NewAtlasDriver(dsn string) (*AtlasDriver, func(), error) {
+	a := strings.Split(dsn, "://")
```

Comment:
use the `url` package and take the `Scheme` value 
https://cs.opensource.google/go/go/+/refs/tags/go1.17.2:src/net/url/url.go;l=359

Reply:
A reference for go-cloud implementation 
https://github.com/google/go-cloud/blob/master/internal/openurl/openurl.go